engine/runtime/src/main/java/org/enso/interpreter/runtime/error/DataflowError.java

```diff
@@ -54,7 +54,10 @@ public final class DataflowError extends AbstractTruffleException implements Ens
   public static DataflowError withDefaultTrace(State state, Object payload, Node location) {
     assert payload != null;
     boolean attachFullStackTrace =
-        state == null ? true : state.currentEnvironment().hasContextEnabled("Dataflow_Stack_Trace");
+        state == null
+            || EnsoContext.get(location)
+                .getExecutionEnvironment()
+                .hasContextEnabled("Dataflow_Stack_Trace");
     if (attachFullStackTrace) {
```

Comment:
There is a performance bug
- #10702

related to code in this method. How does your change affects the benchmarks?